### DISCUSSION TITLE
Support custom breath curves

### DISCUSSION THREAD

**Discussion by bhp1** [edwardkort/WWIDesigner#128] [Whistle Study]
From @CamStLouis: 

> It'd also be nice to select from a couple different "breath curves;" since I find the actual airspeed to rise linearly until the G position, then increase only a little until the 2nd octave D, and then increase in a more quadratic way in the second octave. I can benchmark this by adjusting the imaginary line connecting the green dots, but it would probably improve the accuracy of the predicted values if one could select a more realistic breath curve, or better yet, enter an equation describing exactly what they want (again, relegated to some custom configuration panel so as not to confuse new users.)

More information on real-world breath curves would be helpful.

**Comment by CamStLouis** [comment]
I'd love to collect some data on this, perhaps with a water gauge or something. Embouchure of course still exerts an effect even with whistles, but it would at least be a place to start, jamming a piece of hosing onto the mouthpiece and splicing in a T-junction. I think I even have one somewhere in my brewing equipment.

**Comment by bhp1** [comment]
With reference to recorders, Fletcher and Rossing say, "The general practice is to use a blowing pressure roughly proportional to frequency," attributing it to John Martin's 1994 book, "The Acoustics of the Recorder."  I've seen a graph attributed to Nicholas Montgermont, et al., from ISMA 2007, that shows mean blowing pressure measured for flute players directly proportional to frequency, with a slight dip at the bottom of the second octave.

**Reply by CamStLouis** [reply]
I have created a water gauge! Interesting results. I tested my own high D whistle against a Carbony high D and a Killarney high D, my most frequently used whistles besides my own. I am using mm H2O in a 7mm wide tube. 

[Low G, high G]

Killarney (cylindrical): 5 mm, 30mm - a factor of 6

Carbony (conical): 2.5mm, 20mm - a factor of 8

Barter Loch (cylindrical): 8mm, 35mm - a factor of 4.4

Since the low D is so strong on my whistles (but not the others) I also tested that, and got 5mm, 20mm, a factor of 4.

This confirms my observation that my whistles take much less pressure to play an in-tune second second octave G than many others, and suggests that for pennywhistles at least, assuming a doubling of breath pressure is not an accurate model for what is required to place notes in tune.

If you know of a digital pressure gauge that accepts a single-tube input I'd love to collect more precise data for each note of many different whistles.

**Reply by bhp1** [reply]
Interesting to get results like that.  Haven't got my head around why the pressure ratio would be so high on the Carbony and Killarney.

What I'd really like to get numbers for is air speed of the jet leaving the windway, rather than blowing pressure.  I think this air speed relates more directly to frequency and register than blowing pressure.  WIDesigner has a preliminary model for air speed, which you can see on the [Supplementary Info table](https://github.com/edwardkort/WWIDesigner/wiki/The-Supplementary-Information-Table).

[Torricelli's law](https://en.wikipedia.org/wiki/Torricelli%27s_law) suggests that air speed is proportional to sqrt(pressure).  Looking at a couple of whistles, WIDesigner's air speed model suggests air speed increases a bit more than linearly with frequency, with the second octave about double the speed of the first.  With Torricelli's law, this suggests blowing pressure would increase a bit more than quadratically with frequency,  This is consistent with the results you got for the Barter Loch whistle, but not the Carbony or Killarney.  It isn't consistent with the "roughly proportional to frequency" that I quote above... if Torricelli's law applies.

I'd be interested to see pressure curves for the full range of a whistle, and compare these to WIDesigner's air speed predictions.

**Reply by CamStLouis** [reply]
Interesting ideas - I hadn't heard of Torricelli's law, but it certainly tracks with my experience playing the things. I assume that's the square root of the pressure _differential?_

I agree that pressure alone doesn't provide the most complete view of what's going on; I often must remind my students to focus on blowing _faster_, not "harder," to produce a pleasant second octave tone, though that may simply be a matter of reducing tension in the player's embouchure.

A friend who recently played one of my whistles remarked that the first octave felt like it consumed more air than the second, he described the feeling as "low pressure, but wide in the breath" and the second octave as "higher pressure, but 'narrow' in the breath." 

I just got the finals in this week; I'm waiting on tubing, which has become quite hard to find, but I should be able to send one of these to you soon. WwID really helped with these, after the initial hiccups were figured out.
